### ISSUE TITLE
Upgrade Loki OCI image

### ISSUE THREAD

**Issue by Abuelodelanada** [canonical/loki-k8s-operator#212]
### Enhancement Proposal

In metadata.yaml we have

```yaml
resources:
  loki-image:
    type: oci-image
    description: Loki OCI image "grafana/loki:2.4.1"
    upstream-source: grafana/loki:2.4.1
```

But the last Loki version is [~`2.6.1`~](https://github.com/grafana/loki/releases/tag/v2.6.1) [`2.7.1`](https://github.com/grafana/loki/releases/tag/v2.7.1)

**Comment by simskij** [comment]
We should convert it to Rockcraft while we're at it replicating https://github.com/canonical/mimir-rock/

**Comment by rbarry82** [comment]
Closed with #257
